proto/api.go

```diff
@@ -86,6 +86,12 @@ type Request interface {
 	flags() int
 }
 
+// RequestWithUser is an interface for RPC requests that have a "requested user".
+type RequestWithUser interface {
```

Comment:
Good point. Will move it along with the rest of the code into a hook someplace (most likely security/)